src/java/org/apache/cassandra/index/sai/disk/v1/kdtree/BKDReader.java

```diff
@@ -147,13 +151,63 @@ public IteratorState(DocMapper docMapper, boolean isAscending) throws IOExceptio
             bkdInput.seek(firstLeafFilePointer);
 
             var leafNodeToLeafFP = isAscending ? getLeafOffsets() : getLeafOffsets().descendingMap();
+            if (query != null)
+            {
+                var minLeafFP = findMinLeafFP(query);
+                var maxLeafFP = findMaxLeafFP(query);
+                leafNodeToLeafFP = leafNodeToLeafFP.subMap(minLeafFP, true, maxLeafFP, true);
```

Comment:
Is it worth noting that the `query` handles boundary inclusivity checks? I was curious when I saw the `true`, `true` here.

Reply:
I'll add a comment